nomad/operator_endpoint.go

```diff
@@ -228,6 +228,134 @@ REMOVE:
 	return nil
 }
 
+// TransferLeadershipToServerAddress is used to transfer leadership away from the
+// current leader to a specific target peer. This can help prevent leadership
+// flapping during a rolling upgrade by allowing the cluster operator to target
+// an already upgraded node before upgrading the remainder of the cluster.
+func (op *Operator) TransferLeadershipToServerAddress(args *structs.RaftPeerByAddressRequest, reply *struct{}) error {
```

Comment:
Can we refactor that struct for both the TransferLeadership case and the RemovePeer case, or will that break the API for RemovePeer? I used RemovePeer for inspiration, but it's not strictly necessary that we use the same input structs I guess.

Reply:
Oh, I missed that RemovePeer is where you got this. It's sort of weird that we have two request types there too, so I dug up https://github.com/hashicorp/nomad/pull/2305 and it looks like we originally had by-address only, probably because we only had Raft v2 at the time!

It'd be nice if we had a common `RaftPeerRequest` but to use that for RemovePeer we'd need to implement a new RPC and then deprecate the old one across a major version. I imagine any peer operations are most likely to happen during upgrades, after all, so it'd be really easy to break people's upgrade workflows there. 

I'd suggest creating a common `RaftPeerRequest` here and then we can come back to clean up RemovePeer in a separate PR (which doesn't have to be any time soon; we can leave a TODO for it or open an issue to do on a rainy day).